### DISCUSSION TITLE
Is there a way to print out filename and line number separated by colon?

### DISCUSSION THREAD

**Discussion by rossmacarthur** [BurntSushi/ripgrep#2031] [Q&A] [answered]
Currently the default output shows the filenames and then each case listed underneath. Using `rg -l` prints out only the line numbers, but what I would like is something like the following:

```
path/to/file:107
path/to/file:120
path/to/other/file:83
```

**Comment by BurntSushi** [comment, marked as answer]
> Currently the default output shows the filenames and then each case listed underneath.

You can switch to the standard grep output format with `--no-heading`.

> Using `rg -l` prints out only the line numbers

It does not. ~~It prints matching file paths and the _number of matching lines_.~~ It prints just the matching file paths. If you combine it with `-c/--count`, then it also includes the number of matching lines.

> but what I would like is something like the following

Well, that's what ripgrep does with the `--no-heading` flag, with the exception of also printing the matching contents. So you just have to get rid of the matching contents. Here are a few ways to do that, given a query of `foo`:

```
# Requires modifying the pattern to match the entire line
$ rg '^.*(foo).*$' --no-heading -r ''

# Always works, except when a file path contains a colon.
$ rg 'fn is_empty' --no-heading --line-number | cut -d':' -f1-2
```

You could probably also do something with the `--json` flag and `jq`.

**Reply by rossmacarthur** [reply]
Thanks so much 🙏 

> It does not. It prints matching file paths and the number of matching lines.

Woops, I meant it only prints out the paths.

> Well, that's what ripgrep does with the --no-heading flag,

I must have missed this in the --help. This is sufficient for my use case (I want to be able to ⌘+Click in my editor terminal to open that location)

**Reply by BurntSushi** [reply]
> Woops, I meant it only prints out the paths.

Ah right. My bad. The `--count` flag will include line match counts.

**Reply by MrBrN197** [reply]
> So you just have to get rid of the matching contents. Here are a few ways to do that, 

anyone seeing this after; there is also the `-o`/`--only-matching`

**Reply by lwestby-chwy** [reply]
You can definitely do it with `jq`. Just pass `-s` so it encloses the objects output by `rg --json` into an array, then query:
```sh
 rg --json "foo" file | jq -s '(.[] | select(.type == "match").data.line_number)'
```

**Comment by taliastocks** [comment]
You need to add `-n` to keep the line number when you pipe the output of `rg` now.

**Reply by BurntSushi** [reply]
I think this issue is talking about the `-l/--files-with-matches` flag. In that case, `-n/--line-number` has no effect. You need the `-c/--count` flag to show the number of matches next to each matching file path.
